src/applications/ivc-champva/shared/components/fileUploads/MissingFileList.jsx

```diff
@@ -35,6 +39,66 @@ function alertOrLink(file, entryName, index, fileNameDict = {}) {
   );
 }
 
+// This function produces a single <li> tag that dynamically contains either
+// a link to upload a file, or the text of the file name + additional markup.
+function buildListItems({
+  fileNameMap,
+  file,
+  subset,
+  disableLinks,
+  showFileBullets,
+  entryName,
+  innerIndex,
+  outerIndex,
+  listItemShowNamePrefix,
+}) {
+  const fn = fileNameMap?.[file.name] ?? makeHumanReadable(file.name);
+  const fnStr = typeof fn !== 'string' && fn.name ? fn.name : fn;
+  /*
+    If links are enabled, we show a link to return into the form and upload
+    the missing file. otherwise, we conditionally show the user's name along with
+    the name of the missing file.
+
+    If we have one, we can display an external link to any arbitrary resource that
+    relates to the missing file in question. Usually will be a link to
+    download the PDF. See `REQUIRED_FILES` property in the form-specific 
+    constants.js file to set links
+   */
+  return file.required === subset &&
+    (disableLinks ? file.uploaded === false : true) ? (
+    <li
+      key={file.name + file.uploaded + innerIndex}
+      className="vads-u-margin-y--1"
+    >
+      {!disableLinks ? (
+        <>
+          {showFileBullets ? fnStr : null}
+          <br />
+          {alertOrLink(file, entryName, outerIndex, fileNameMap)}
+        </>
+      ) : (
+        <>
+          {listItemShowNamePrefix ? `${entryName}’s` : ''} {fnStr}
+          {fn?.href !== undefined &&
+            fn?.linkText !== undefined && (
+              <>
+                <br />
+                <a
+                  className="vads-u-margin-left--3"
+                  href={fn.href}
+                  rel="noreferrer"
+                  target="_blank"
+                >
+                  {fn.linkText}
+                </a>
+              </>
+            )}
+        </>
+      )}
```

Comment:
Main point of this chunk is to render an extra `<a>` if we have one present in the `REQUIRED_FILES` element we're looking at right now. 

This whole function was pulled out of the `MissingFileList` component to help break things up/be a bit more modular.